l10n_es_intrastat_report/migrations/16.0.1.3.1/post-migration.py

```diff
@@ -0,0 +1,113 @@
+        "intrastat_product.intrastat_transaction_32",
+    ),
+    (
+        "l10n_es_intrastat_report.intrastat_transaction_33",
+        "intrastat_product.intrastat_transaction_33",
+    ),
+    (
+        "l10n_es_intrastat_report.intrastat_transaction_34",
+        "intrastat_product.intrastat_transaction_34",
+    ),
+    (
+        "l10n_es_intrastat_report.intrastat_transaction_41",
+        "intrastat_product.intrastat_transaction_41",
+    ),
+    (
+        "l10n_es_intrastat_report.intrastat_transaction_42",
+        "intrastat_product.intrastat_transaction_42",
+    ),
+    (
+        "l10n_es_intrastat_report.intrastat_transaction_51",
+        "intrastat_product.intrastat_transaction_51",
+    ),
+    (
+        "l10n_es_intrastat_report.intrastat_transaction_52",
+        "intrastat_product.intrastat_transaction_52",
+    ),
+    (
+        "l10n_es_intrastat_report.intrastat_transaction_71",
+        "intrastat_product.intrastat_transaction_71",
+    ),
+    (
+        "l10n_es_intrastat_report.intrastat_transaction_72",
+        "intrastat_product.intrastat_transaction_72",
+    ),
+    (
+        "l10n_es_intrastat_report.intrastat_transaction_80",
+        "intrastat_product.intrastat_transaction_80",
+    ),
+    (
+        "l10n_es_intrastat_report.intrastat_transaction_91",
+        "intrastat_product.intrastat_transaction_91",
+    ),
+    (
+        "l10n_es_intrastat_report.intrastat_transaction_99",
+        "intrastat_product.intrastat_transaction_99",
+    ),
+]
+_xmlids_unlink_records = [
+    "l10n_es_intrastat_report.intrastat_transaction_13",
+    "l10n_es_intrastat_report.intrastat_transaction_14",
+    "l10n_es_intrastat_report.intrastat_transaction_19",
+    "l10n_es_intrastat_report.intrastat_transaction_29",
+    "l10n_es_intrastat_report.intrastat_transaction_44",
+    "l10n_es_intrastat_report.intrastat_transaction_45",
+    "l10n_es_intrastat_report.intrastat_transaction_54",
+    "l10n_es_intrastat_report.intrastat_transaction_55",
+    "l10n_es_intrastat_report.intrastat_transaction_70",
+]
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    """Merge records from intrastat.transaction.
+    Existing records from l10n_en_intrastat_report will be merged into
+    intrastat_product records.
+    Remove obsolote transactions (previously archived)."""
+    for item in _xmlids_merge_records:
+        old_record = env.ref(item[0], raise_if_not_found=False)
+        new_record = env.ref(item[1], raise_if_not_found=False)
+        if old_record and new_record:
+            openupgrade_merge_records.merge_records(
+                env=env,
+                model_name="intrastat.transaction",
+                record_ids=old_record.ids,
+                target_record_id=new_record.id,
+                method="sql",
+            )
+    for item in _xmlids_unlink_records:
+        record = env.ref(item, raise_if_not_found=False)
+        if record:
+            openupgrade.safe_unlink(record)
```

Comment:
I think it's better to use: https://github.com/OCA/openupgradelib/blob/69f155a85b12a8d4f30d8a3c5696046e4c4626fa/openupgradelib/openupgrade.py#L3194C36